core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -565,8 +565,9 @@ private[spark] class BlockManager(
             // Give up trying anymore locations. Either we've tried all of the original locations,
             // or we've refreshed the list of locations from the master, and have still
             // hit failures after trying locations from the refreshed list.
-            throw new BlockFetchException(s"Failed to fetch block after" +
-              s" ${totalFailureCount} fetch failures. Most recent failure cause:", e)
+            logError(s"Failed to fetch block after $totalFailureCount fetch failures." +
```

Comment:
If a block is not found at a particular location, I take it that's a non-fatal error ? In that case, should we really be logging "error" if the block is not found ? Or do we expect the block to be found *somewhere* remotely during the normal course of execution ?